src/main/java/io/vertx/core/eventbus/impl/clustered/ConnectionHolder.java

```diff
@@ -61,32 +62,35 @@
     client = new NetClientImpl(eventBus.vertx(), clientOptions, false);
   }
 
-  synchronized void connect() {
-    if (connected) {
-      throw new IllegalStateException("Already connected");
-    }
-    client.connect(serverID.port, serverID.host, res -> {
-      if (res.succeeded()) {
-        connected(res.result());
-      } else {
-        log.warn("Connecting to server " + serverID + " failed", res.cause());
-        close(res.cause());
+  void connect() {
+    synchronized (this) {
+      if (connected) {
+        throw new IllegalStateException("Already connected");
       }
-    });
+    }
+    eventBus.vertx().getClusterManager().getNodeInfo(remoteNodeId)
```

Comment:
looking at this, shouldn't `getNodeInfo` actually fails the future instead of completing the future will null ? that would be more clean and avoid an unecessary flatMap operation only for this purpose

Reply:
Done